### DISCUSSION TITLE
Can someone ELI5 the concept of a mask in SHAP when working with tabular data?

### DISCUSSION THREAD

**Discussion by jolespin** [shap/shap#2790] [Q&A]
My understanding of a mask in arrays is to have boolean values matching the shape of a query array (n, m) where you would mask the query and perform an operation. For example, this operation to sum only the positive values in an array:

```python
import numpy as np
query = np.random.normal(size=(5,10))
mask = query > 0
np.sum(query[mask], axis=1)
```

1. Is the concept of masking in `SHAP` similar, the same, or completely different? I'm talking exclusively about tabular data. 
2. How do you decide on a mask to make the process reproducible and not subjective? 

Let's say you have the following conditions:
    * Binary classification: Healthy, Disease
    * Multiclass classification: Healthy, Disease_A, Disease_B (mutually exclusive, can only be one per sample)

3. In a lot of the examples, they are masking a training set and then calculating SHAP values on the test set.  This seems to be dependent on the random seed.  Would it be better to use cross-validation and calculate a distribution of SHAP values per feature?  

For example, like this: 

```python
for training_index, testing_index in KFold(n_splits=3, random_state=0, shuffle=True).split(X, y):
    X_training = X.iloc[training_index]
    X_testing = X.iloc[testing_index]
    explainer = shap.KernelExplainer(clf.predict_proba, X_training)
    shap_values = explainer.shap_values(X_testing)
```

4. Last question, how can I calculate a "final" SHAP value for my dataset given all of the data?  Either a single vector that has the total predictive capacity of a feature or an array (m features, c classes).
